### DISCUSSION TITLE
Sinatra Namespace and sinatra.route

### DISCUSSION THREAD

**Discussion by ddanurwenda** [sinatra/sinatra#1927] [Q&A] [answered]
Hi, I found this issue where the `env['sinatra.route']` uses different format if using namespace. Specifically, it will wrap param in curly brackets instead of the original colon.

``` ruby
require 'sinatra'

namespace '/foo/:foo_id' do
  get '/bar/:bar_id' do
    p env['sinatra.route'] # will print GET /foo/{foo_id}/bar/{bar_id}
  end
end

get '/foo/:foo_id/baz/:baz_id' do
  p env['sinatra.route'] # will print GET /foo/:foo_id/baz/:baz_id
end
```

Is there a way to config the namespace s.t it won't affect the `sinatra.route`? Thanks

**Comment by dentarg** [comment]
I don't think you can configure anything so this wont happen... but that is just my hunch, I haven't looked into the code. Do you want to investigate it @ddanurwenda?

**Comment by ddanurwenda** [comment, marked as answer]
A quick investigation tells me it's the default behavior of mustermann when [concatenating](https://github.com/sinatra/sinatra/blob/5f4dde19719505989905782a61a19c545df7f9f9/sinatra-contrib/lib/sinatra/namespace.rb#L344) two sinatra patterns (with colon) that results one with curly braces. https://github.com/sinatra/mustermann/blob/377855be0297025752f024601d4eed33389aed40/mustermann/spec/concat_spec.rb#L7-L11.
